File: backend/Domain/Entities/Quizzes/Question.cs
using KvizHub.Domain.Enums;

namespace KvizHub.Domain.Entities.Quizzes;

public class Question
{
    public int Id { get; set; }
    public string Text { get; set; } = default!;
    public QuestionType Type { get; set; }
    public int Points { get; set; } = 1;

    public int QuizId { get; set; }
    public Quiz Quiz { get; set; } = default!;

    public ICollection<AnswerOption> AnswerOptions { get; set; } = new List<AnswerOption>();
    public ICollection<UserAnswer> UserAnswers { get; set; } = new List<UserAnswer>();
}

File: frontend/kviz-hub-ui/src/models/Question.ts
export type QuestionType = "SingleChoice" | "MultipleChoice" | "TrueFalse" | "FillInBlank";

export interface AnswerOption {
  id: number;
  text: string;
  isCorrect?: boolean;
}

export interface Question {
  id: number;
  text: string;
  type: QuestionType;
  points: number;
  quizId: number;
  answerOptions: AnswerOption[];
}